src/main/webapp/maps_geolocation.js

```diff
@@ -0,0 +1,176 @@
+const NO_LOCATION_PERMISSION_MSG = "Please allow location permission!.";
+const NO_LOCATION_INFORMATION_MSG = "Location information is unavailable.";
+const REQUEST_TIMEOUT_MSG = "The request to get user location timed out.";
+const UNKNOWN_ERROR_MSG = "An unknown error occurred while geolocating.";
+
+const GEOLOCATION_IMG_ID = "enableGeolocationIcon";
+const ENABLE_GEOLOCATION_TITLE = "Enable geolocation";
+const DISABLED_GEOLOCATION_IMG_SRC = "./img/geolocation.svg"
+const DISABLE_GEOLOCATION_TITLE = "Disable geolocation";
+const ENABLE_GEOLOCATION_IMG_SRC = "./img/geolocation_active.svg";
+
+const GO_TO_MY_LOCATION_IMG_ID = "goToMyLocationIcon";
+const GO_TO_MY_LOCATION_TITLE = "Go to my location";
+
+/**
+ * Adds a button to the map which turns on/off geolocation.
+ * Geolocation can be battery-consuming, and the user should be able
+ * to tund it off.
+ * Remark that the audio-guide creation process doesn't require the user's location at all.
+ */
+function addEnableGeolocationControl(map) {
+  const geolocationControlDiv =
+      createControlDiv(ENABLE_GEOLOCATION_TITLE, DISABLED_GEOLOCATION_IMG_SRC, GEOLOCATION_IMG_ID);
+  geolocationControlDiv.index = 1;
+  map.controls[google.maps.ControlPosition.RIGHT_BOTTOM].push(geolocationControlDiv);
+  trackUser = false;
+  geolocationControlDiv.addEventListener("click",
+          event => geolocationControlEvent(map, geolocationControlDiv));
+}
+
+/**
+ * When the geolocation button gets clicked, its status is toggled.
+ * If geolocation is on, then each time the user's loctaion changes,
+ * and event will be triggered,
+ * and the currentLocationMarker and currentlocation will be reset
+ */
+function geolocationControlEvent(map, geolocationControlDiv) {
+  var img = document.getElementById(GEOLOCATION_IMG_ID);
+  trackUser = !trackUser;
+  if (trackUser) {
+    geolocationControlDiv.title = DISABLE_GEOLOCATION_TITLE;
+    img.src = ENABLE_GEOLOCATION_IMG_SRC;
+    if (navigator.geolocation) {
+      watchPositionId = navigator.geolocation.watchPosition(
+          position => {
+            saveCurrentlocation(position, map);
+            showCurrentLocationMarker(map)
+          },
+          error => watchPositionError(error));
+    } else {
+      alert(NO_GEOLOCATION_SUPPORT_MSG);
+    }
+  } else {
+    navigator.geolocation.clearWatch(watchPositionId);
+    removeCurrentLocationMarker();
+    img.src = DISABLED_GEOLOCATION_IMG_SRC;
+    geolocationControlDiv.title = ENABLE_GEOLOCATION_TITLE;
+  }
+}
+
+/**
+ * Adds a button to the map which lets the user center the map around their current location.
+ */
+function addGoToMyLocationControl(map) {
+  const myLocationControlDiv =
+      createControlDiv(GO_TO_MY_LOCATION_TITLE,
+              "./img/my_location.svg", GO_TO_MY_LOCATION_IMG_ID);
+  myLocationControlDiv.index = 1;
+  map.controls[google.maps.ControlPosition.RIGHT_BOTTOM].push(myLocationControlDiv);
+  trackUser = false;
```

Comment:
Do we need to change the trackUser state here? To me it sounds like it could be done only in the `geolocationControlEvent()` method

Reply:
Removed. I actually added the other initialization later, and forgot about this one.